DOCUMENTATION: Clarify experience gain of sleeves

bupjae opened the pull request:
* Shock 100 is full shock and gain no experience, while shock 0 is no shock and gain full experience. The amount of experience should be (100-X)% rather than X%.

* https://github.com/bitburner-official/bitburner-src/pull/211 applies shock penalty on receiving sleeve, which is also needed to explain on this document.

bupjae committed:
DOCUMENTATION: Clarify experience gain of sleeves

* Shock 100 is full shock and gain no experience, while shock 0 is no shock and gain full experience. The amount of experience should be (100-X)% rather than X%.

* https://github.com/bitburner-official/bitburner-src/pull/211 applies shock penalty on receiving sleeve, which is also needed to explain on this document.

d0sboots commented on src/Documentation/doc/advanced/sleeves.md:
```diff
@@ -52,8 +52,12 @@ Sleeve shock slowly decreases over time.
+The player’s original host consciousness earn `Y%` of the experience that the sleeve gained, or `(100-X)*Y/100 %` of the normal experience amount.
+
+Other sleeves also earn the experience. However, their own shock affect this gain.
+Let `Z` be the receiving sleeve's shock, then it earns `(100-Z) %` of the experience the working sleeve synchronizing, or `(100-X)*Y*(100-Z)/10000 %` of the normal experience amount.
```
Oof. This is really suffering from how percentages don't really work when multiplied/divided.

I would suggest changing this section to use fractions (0-1) instead of percentages (0-100%). Then multiplication just works naturally. You could also just say that "\`X\` = 1 - the sleeve's shock", and then you don't need to write (1-X) everywhere.

bupjae replied:
Modified to use fraction ratio (range 0~1) for formular instead of percentage.
Please have another look.

bupjae committed:
Use fraction ratio (range 0~1) instead of percentage (range 0~100) for formulas

Formular based on percentage don't work well when we need to multiply / divide them.

bupjae committed:
small typo

d0sboots commented on src/Documentation/doc/advanced/sleeves.md:
```diff
@@ -51,9 +51,13 @@ Shock affects the amount of experience earned by the sleeve.
-Let `X` be the sleeve's shock and `Y` be the sleeve's synchronization.
-When the sleeve earns experience by performing a task, the sleeve gains `X%` of the amount of experience normally earned by the task.
-The player’s original host consciousness and all of the player's other sleeves earn `Y%` of the experience that the sleeve gained, or `X\*Y %` of the normal experience amount.
+Let `X` be the sleeve's recover ratio (calculated as `(100 - shock) / 100`), and `Y` be the sleeve's synchronization ratio (calculated as `synchronize / 100`).
```
```suggestion
Let `X` be the sleeve's recovery ratio (calculated as `(100 - shock) / 100`), and `Y` be the sleeve's synchronization ratio (calculated as `synchronize / 100`).
```

d0sboots commented on src/Documentation/doc/advanced/sleeves.md:
```diff
@@ -51,9 +51,13 @@ Shock affects the amount of experience earned by the sleeve.
+Let `X` be the sleeve's recover ratio (calculated as `(100 - shock) / 100`), and `Y` be the sleeve's synchronization ratio (calculated as `synchronize / 100`).
+When performing a task would give you `A` experience normally, the sleeve gains `A * X` experience instead.
+
+The player’s original host consciousness earn fraction of the experience sleeve gained, namely `A * X * Y`.
+
+Other sleeves also earn the experience. However, their own shock affect this gain.
+Let `Z` be the receiving sleeve's recover ratio. It earns `A * X * Y * Z` experience.
```
```suggestion
The player’s original host consciousness earns a fraction of the experience the sleeve gained, namely `A * X * Y`.

Other sleeves also earn the experience. However, their own shock affects this gain.
Let `Z` be the receiving sleeve's recovery ratio. It earns `A * X * Y * Z` experience.
```

d0sboots reviewed (commented):
Just some copy-editing nits, looks good.

bupjae committed:
Apply suggestions from code review

Co-authored-by: David Walker <d0sboots@gmail.com>

bupjae commented:
Applied suggestions.
